### DISCUSSION TITLE
Are accounts deployed using hardhat OpenZeppelin in starknet cairo are upgradable contracts?

### DISCUSSION THREAD

**Discussion by zeapherine** [OpenZeppelin/cairo-contracts#577] [Q&A] [answered]
Hey, I wanted to know if the accounts deployed using hardhat OpenZeppelin in starknet cairo are upgradable contracts.

I am inquiring about this, since, After the starknet regenesis, if i am not mistaken, to migrate the already deployed contract to v1 of cairo, the existing contracts will need to be a upgradable contracts. So i just wanted to make sure that the accounts deployed using hardhat Openzeppelin are upgradable as well, since the admin of the contract already deployed at the moment is the account deployed using hardhat OpenZeppelin.

Thanks.

**Comment by martriay** [comment, marked as answer]
This seems like a hardhat question rather than an OpenZeppelin one. I would assume that unless they offer a modified version, no, they're not upgradeable. You'll have to write a custom Account extended with the [upgrades library](https://docs.openzeppelin.com/contracts-cairo/0.6.1/proxies#upgrades_library_api).

**Reply by zeapherine** [reply]
Thank You.
